tests/test_modeling_common.py

```diff
@@ -3385,6 +3385,9 @@ def test_eager_matches_sdpa_inference(self, torch_dtype: str):
         if torch_device == "cpu" and torch_dtype == "float16":
             self.skipTest("float16 not supported on cpu")
 
+        if torch_device == "cuda" and torch_dtype == "bfloat16" and not torch.cuda.is_bf16_supported():
+            self.skipTest("the gpu used does not support bfloat16 data type")
```

Comment:
There have been recent efforts to make transformers tests device agnostic - #25654. 

In this case, we want to use something like `is_torch_bf16_available_on_device` here to check and skip here